pmd-javascript/src/main/java/net/sourceforge/pmd/lang/ecmascript/ast/EcmascriptParser.java

```diff
@@ -19,28 +19,31 @@
 import org.mozilla.javascript.ast.ErrorCollector;
 import org.mozilla.javascript.ast.ParseProblem;
 
+import net.sourceforge.pmd.internal.util.AssertionUtil;
+import net.sourceforge.pmd.lang.ParserOptions;
 import net.sourceforge.pmd.lang.ast.ParseException;
-import net.sourceforge.pmd.lang.ecmascript.EcmascriptParserOptions;
 
-public class EcmascriptParser {
-    protected final EcmascriptParserOptions parserOptions;
+public final class EcmascriptParser implements net.sourceforge.pmd.lang.Parser {
+    private final int esVersion;
+    private final String suppressMarker;
 
-    private Map<Integer, String> suppressMap;
-    private String suppressMarker = "NOPMD"; // that's the default value
+    public EcmascriptParser(int version, String suppressMarker) {
+        this.esVersion = version;
+        this.suppressMarker = AssertionUtil.requireParamNotNull("suppression marker", suppressMarker);
```

Comment:
Why not Objects.requireNonNull? Ok, it has a little bit additional support for building the NPE message.....